### PR TITLE
docs: clarify CSI plugin compatibility

### PR DESCRIPTION
**Pull request by tgross**
Nomad is generally compliant with the CSI specification for Container
Orchestrators (CO), except for unimplemented features. However, some storage
vendors have built CSI plugins that are not compliant with the specification or
which expect that they're only deployed on Kubernetes. Nomad cannot vouch for
the compatibility of any particular plugin, so clarify this in the docs.

### CODE REVIEW

File: website/content/docs/concepts/plugins/csi.mdx [hunk 1]
```diff
@@ -30,9 +30,14 @@ plugin. Jobs can claim storage volumes from AWS Elastic Block Storage
...
 Nomad scheduler will be aware of volumes created by CSI plugins and
 schedule workloads based on the availability of volumes on a given
-Nomad client node. A list of available CSI plugins can be found in the
-[Kubernetes CSI documentation][csi-drivers-list]. Any of these plugins
-should work with Nomad out of the box.
+Nomad client node.
+
+A list of available CSI plugins can be found in the [Kubernetes CSI
+documentation][csi-drivers-list]. Any of these plugins should work with Nomad
+out of the box, but it's some plugin vendors have implemented their plugins in a
```
Comment (DerekStrickland) [inline website/content/docs/concepts/plugins/csi.mdx:37]:
```suggestion
However, it is possible a plugin vendor has implemented their plugin 
```

File: website/content/docs/concepts/plugins/csi.mdx [hunk 2]
```diff
@@ -30,9 +30,14 @@ plugin. Jobs can claim storage volumes from AWS Elastic Block Storage
 (EBS) volumes, GCP persistent disks, Ceph, Portworx, vSphere, etc. The
 Nomad scheduler will be aware of volumes created by CSI plugins and
 schedule workloads based on the availability of volumes on a given
-Nomad client node. A list of available CSI plugins can be found in the
-[Kubernetes CSI documentation][csi-drivers-list]. Any of these plugins
-should work with Nomad out of the box.
+Nomad client node.
+
+A list of available CSI plugins can be found in the [Kubernetes CSI
+documentation][csi-drivers-list]. Any of these plugins should work with Nomad
```
Comment (DerekStrickland) [inline website/content/docs/concepts/plugins/csi.mdx:36]:
```suggestion
documentation][csi-drivers-list]. Spec-compliant plugins should work with Nomad.
```

**Review by DerekStrickland — COMMENTED** [review]
How do you feel about these suggestions?

File: website/content/docs/concepts/plugins/csi.mdx (hunk 1 above)
Comment (tgross) [reply website/content/docs/concepts/plugins/csi.mdx:37]: "it's some"? What was I even typing there? 😊

**Review by DerekStrickland — APPROVED** [review]
LGTM

**Review by lgfa29 — APPROVED** [review]

**Pull request merged**